src/libAtomVM/otp_socket.c

```diff
@@ -284,12 +470,28 @@ static inline int get_protocol(GlobalContext *global, term protocol_term, bool *
 
 static inline term make_error_tuple(term reason, Context *ctx)
```

Comment:
We should probably have a comment here to the effect that you MUST return the result of this function from a Nif implementation; otherwise the function has side effect on ctx that may go unnoticed.